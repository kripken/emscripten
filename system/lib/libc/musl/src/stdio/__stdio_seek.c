#include "stdio_impl.h"
#ifndef __EMSCRIPTEN__
#include <unistd.h>
#endif

off_t __stdio_seek(FILE *f, off_t off, int whence)
{
#ifdef __EMSCRIPTEN__
	off_t ret;
	if (__wasi_syscall_ret(__wasi_fd_seek(f->fd, off, whence, &ret)))
		ret = -1;
	return ret;
#else
	return __lseek(f->fd, off, whence);
#endif // __EMSCRIPTEN__
}
