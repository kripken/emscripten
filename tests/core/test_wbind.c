#include <stdint.h>
#include <assert.h>
#include <stdio.h>

extern void test_wbind_iijdf(int (*test_func)(int, uint64_t, double, float));
extern void test_wbind_ji(uint64_t (*test_func)(int));

int test_func_iijdf(int a, uint64_t b, double d, float f)
{
	printf("a: %d\n", a);
	printf("b: %llx\n", b);
	printf("d: %g\n", d);
	printf("f: %f\n", f);
	return 42;
}

uint64_t test_func_ji(int a)
{
	printf("a: %d\n", a);
	return ((uint64_t)a << 32) | (uint64_t)a;
}

int main()
{
	test_wbind_iijdf(test_func_iijdf);
	test_wbind_ji(test_func_ji);
}
