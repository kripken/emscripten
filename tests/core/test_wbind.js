mergeInto(LibraryManager.library, {
  test_wbind_iijdf__deps: ['$wbind', '$wbindArray'],
  test_wbind_iijdf: function(funcPtr) {
    var ret = wbind(funcPtr)(1, BigInt(0x300000002), 4.2, 5.3);
    console.log(ret);

    ret = wbindArray(funcPtr)([1, BigInt(0x300000002), 4.2, 5.3]);
    console.log(ret);
  },

  test_wbind_ji__deps: ['$wbind', '$wbindArray'],
  test_wbind_ji: function(funcPtr) {
    var ret = wbind(funcPtr)(1);
    console.log(ret);

    ret = wbindArray(funcPtr)([1]);
    console.log(ret);
  }
});
