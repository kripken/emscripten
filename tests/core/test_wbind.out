a: 1
b: 300000002
d: 4.2
f: 5.300000
42
a: 1
b: 300000002
d: 4.2
f: 5.300000
42
a: 1
4294967297n
a: 1
4294967297n